sentry/src/main/java/io/sentry/MainEventProcessor.java

```diff
@@ -123,5 +130,14 @@ private void processNonCachedEvent(final @NotNull SentryEvent event) {
         event.setThreads(sentryThreadFactory.getCurrentThread());
       }
     }
+    if (options.isSendDefaultPii()) {
+      if (event.getUser() == null) {
+        final User user = new User();
+        user.setIpAddress(DEFAULT_IP_ADDRESS);
+        event.setUser(user);
+      } else if (event.getUser().getIpAddress() == null) {
+        event.getUser().setIpAddress(DEFAULT_IP_ADDRESS);
```

Comment:
You probably want to _undo_ this in the `spring` integration, since `{{auto}}` there will give the IP of the server.

Like check for default and set null instead.

Reply:
In such case wouldn't it be better if we make it an opt in which is pre-configured in Android? Undoing it in Spring looks for me like a workaround.

Reply:
@maciejwalkowiak That means desktop apps would have to opt-in too.
I think it's fair to say: It's enabled by default but you can opt-out. And we know better in some integrations to opt out already.

We need to document this behavior. Similar to how the default of another option is documented here:
https://github.com/getsentry/sentry-docs/pull/2815

Reply:
if we check for `{{auto}}` and set it to null if `isSendDefaultPii` is disabled would be a breaking change btw which is fine as 4.0 is a major bump, but yeah we'd need to document.

Also, it could be written `{{ auto }}` with spaces, so we'd need to trim spaces off if we want to compare that properly as well, otherwise, it'd not be a match and IPs will be set by the server.

Reply:
this is breaking the Android user, see:

https://github.com/getsentry/sentry-java/blob/main/sentry-android-core/src/main/java/io/sentry/android/core/DefaultAndroidEventProcessor.java#L151-L154

now the `user` will never be null, we need to fix the `if` condition in the `DefaultAndroidEventProcessor` class

Reply:
Undoing in Spring integration ✅
Updating Android code ✅

Reply:
this has been merged but there's a bug here.

`{{ auto }}` is valid
`{{auto}}` is also valid

previously `{{auto}}` was not valid and its been fixed on the backend, so people were setting `{{ auto }}` manually, those people, it won't work as the comparison is not trimming off the spaces.

Reply:
Got it. Will be fixed in next PR.